crates/ty_python_semantic/src/types/display.rs

```diff
@@ -467,30 +659,35 @@ impl Display for DisplayRepresentation<'_> {
                             settings: self.settings.clone(),
                         };
 
-                        write!(
-                            f,
-                            "bound method {instance}.{method}{type_parameters}{signature}",
-                            method = function.name(self.db),
-                            instance = self_ty.display_with(self.db, self.settings.singleline()),
-                            type_parameters = type_parameters,
-                            signature = signature
-                                .bind_self(self.db, Some(typing_self_ty))
-                                .display_with(self.db, self.settings.clone())
-                        )
+                        f.write_str("bound method ")?;
+                        self_ty
+                            .display_with(self.db, self.settings.singleline())
+                            .fmt_detailed(f)?;
+                        f.write_char('.')?;
+                        f.with_detail(TypeDetail::Type(self.ty))
+                            .write_str(function.name(self.db))?;
+                        type_parameters.fmt_detailed(f)?;
+                        signature
+                            .bind_self(self.db, Some(typing_self_ty))
+                            .display_with(self.db, self.settings.clone())
+                            .fmt_detailed(f)
                     }
                     signatures => {
                         // TODO: How to display overloads?
                         if !self.settings.multiline {
                             f.write_str("Overload[")?;
                         }
                         let separator = if self.settings.multiline { "\n" } else { ", " };
-                        let mut join = f.join(separator);
+                        let mut is_first = true;
                         for signature in signatures {
-                            join.entry(
-                                &signature
-                                    .bind_self(self.db, Some(typing_self_ty))
-                                    .display_with(self.db, self.settings.clone()),
-                            );
+                            if !is_first {
+                                f.write_str(separator)?;
+                            }
+                            is_first = false;
+                            signature
+                                .bind_self(self.db, Some(typing_self_ty))
+                                .display_with(self.db, self.settings.clone())
+                                .fmt_detailed(f)?;
```

Comment:
With apologies to `join` enjoyers but I had to hand-roll this gunk everywhere to invoke fmt_detailed